starter_code/src/movies.js

```diff
@@ -1,15 +1,81 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+// const movies2 = [...movies];
+
+// movies2.sort((a, b) => a.year - b.year);
+
+function orderByYear(arr) {
+  const arr2 = [...arr]; //preserve the original one
+  //   movies2.sort((a, b) => a.year - b.year);
+  arr2.sort(function(a, b) {
+    if (a.year < b.year) {
+      return -1;
+    } else if (a.year > b.year) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+  return arr2; //return the sorted array
+}
+
+// console.log(moviesSorted);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+  const arr2 = arr.filter(el => {
+    return el.genre.includes("Drama") && el.director === "Steven Spielberg";
+  });
+
+  console.log(`Movies ${arr2}`);
+  return arr2.length; //return the filtered array
+}
+
+howManyMovies(movies);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(arr) {
+  const arr2 = [...arr];
+  const arrSorted = arr2.sort(function(a, b) {
+    if (a.title < b.title) {
+      return -1;
+    } else if (a.title > b.title) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+
+  const arr20 = arrSorted.slice(0, 20);
+
+  const titles20 = arr20.map(function(movie) {
+    return movie.title;
+  });
+  return titles20;
+}
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(arr) {
+  const arr2 = [...arr];
+  const ratesArr = arr2.map(function(movie) {
+    return movie.rate;
+  });
+
+  let rateTotal = ratesArr.reduce((sum, movie) => {
+    return sum + movie.rate;
```

Comment:
Here, instead of the movie element, you already have the rate of the movie (because you created a new ratesArr above with only the rates).
So you would have:
 let rateTotal = ratesArr.reduce((sum, rate) => {
    return sum + rate;
}

In addition, one of the tests requires to return a result even if one of the movies does not have rate, so you can add a check inside the reduce method to check if "rate" exists or not:
if(!movie.rate) { return sum; }
else { return sum + movie.rate; }